caluma/core/validations.py

```diff
@@ -0,0 +1,66 @@
+import inspect
+from functools import wraps
+
+from django.core.exceptions import ImproperlyConfigured
+
+from .collections import list_duplicates
+
+
+def validation_for(mutation):
+    """Decorate function to overwriting validation of specific mutation."""
+
+    def decorate(fn):
+        @wraps(fn)
+        def _decorated(self, mutation, data, info):
+            return fn(self, mutation, data, info)
+
+        _decorated._validation = mutation
+        return _decorated
+
+    return decorate
+
+
+class BaseValidation(object):
+    """Basic validation class to be extended by any validation implementation.
+
+    In combination with the decorators `@validation_for` a custom validation class
+    can define validation on basis of mutations and its subclasses.
```

Comment:
```suggestion
    can define validations both for specific mutations as well as the base `Mutation` class.
```

Reply:
Not so sure on this change as it makes it inconsistent with other extension points and basically just repeats what the code already way more simply states.